### PR TITLE
Update hacking.rst

### PR DESCRIPTION
**Pull request by ashtongreen**
Added notes on how Hacking Works.

### CODE REVIEW

**Review by phyzical — APPROVED** [review]

File: doc/source/basicgameplay/hacking.rst [hunk 1]
```diff
@@ -107,3 +107,10 @@ starting security, rounded to the nearest integer. To be more precise::
 
 This means that a server's security level will not fall below this
 value if you are trying to weaken() it.
+
+Backdoors
+^^^^^^^^^
+Servers that have been hacked can have backdoors installed. These backdoors
```
Comment (phyzical) [inline doc/source/basicgameplay/hacking.rst:113]: maybe also mention something about the fact that backdooring a server allows direct connection via `connect`

Comment (ashtongreen) [reply doc/source/basicgameplay/hacking.rst:113]: Didn't know that was a thing. I'll add it now.

Comment (phyzical) [reply doc/source/basicgameplay/hacking.rst:113]: neither did i till the other day 😆

File: doc/source/basicgameplay/hacking.rst [hunk 2]
```diff
@@ -107,3 +107,14 @@ starting security, rounded to the nearest integer. To be more precise::
...
+will provide you with a benefit; the services may be cheaper, penalties may
+be reduced or there may be other results. Honeypots exist and will let factions
+know when you have succeeded at backdooring their system. Once you have a 
+backdoor installed, you can connect directly that a server.
```
Comment (ApamNapat) [inline doc/source/basicgameplay/hacking.rst:117]:
```suggestion
backdoor installed, you can connect to that server directly. 
```

**Review by phyzical — APPROVED** [review]
LGTM 👍

**Comment by TheRealMaxion** [comment]
Backdooring a server does not require using hack() on the server first, it just requires root access and enough hacking skill. So the wording on this could use improvement.

**Comment by ashtongreen** [comment]
> Backdooring a server does not require using hack() on the server first, it just requires root access and enough hacking skill (and being connected to the server). So the wording on this could use improvement.

That should resolve the issue.

**Pull request merged**